### PR TITLE
Updates and corrections

### LINKED ISSUE #48: wet-to-dry conversion not CCPP compliant
The routines, `wet_to_dry_run` and `dry_to_wet_run` are not CCPP compliant because they use the same variable (and 
therefore same standard name) for both the wet and the dry state.

In addition, these routines use `optional` arguments which are not supported by the CCPP Framework.

The output variables of these routines need to be separate from the associated input state variable.

To avoid the use of `optional` arguments, separate routines need to be provided for water vapor and standard water species. In addition, to allow conversion of other constituents and groups of constituents, a routine that takes an array of constituents as input should be provided. One issue here is how to define the input and output arrays. Currently, we do not have a naming scheme for a constituent array that would distinguish 'dry' from 'wet' (or 'moist', are we even going to support 'moist'?&dagger;) and there will always elements of these arrays which are not convertible between 'wet' and 'dry'.

One possibility is to try and use the framework to generate code to keep track of the state of variables inside a constituent array. A companion constituent property array could be required that would keep track of the state of each of the variables, however, there is no easy way to track variables which are separate indices into the constituent array. Ideas?

&dagger; In CAM, we determine 'wet' via a namelist variable specifying the water species which participate in the 'wet air mass'. However, the [CCPP standard name rules](https://github.com/ESCOMP/CCPPStandardNames/blob/main/StandardNamesRules.rst) distinguished between `_wrt_moist_air` and `_wrt_total_mass` (see rule 5). My inclination is to go with `_wrt_total_mass` as that is how we will usually use the variables. Thoughts?

### LINKED ISSUE #51: The variable "ccpp_error_message" needs to have units of "none"
Currently all instances of `ccpp_error_message` in this repo have units of one (`1`).  However, according to the official CCPP standard names dictionary `ccpp_error_message` should have units of `none`.   

This matters because without the correct units the latest version of the CCPP-framework will error out.

### PR DESCRIPTION
**Pull request by gold2718**
Update metadata
Remove unsupported optional `metadata` keyword
Add constituent info to metadata (i.e., `advected` property)
Rewrite wet_to_dry_to_wet converters to be proper CCPP
Add qneg_namelist.xml

fixes #51 
fixes #48

### CODE REVIEW

**Comment by gold2718** [comment]
@nusbaume, Kessler still has an issue because the CCPP Framework is getting confused by multiple schemes in a file (the converters) and is sometimes getting the wrong module for an input variable (so physics_inputs.F90 does not compile).
I will work on a fix for that.

**Review by peverwhee — APPROVED** [review]
looks good to me

**Review by nusbaume — CHANGES_REQUESTED** [review]
Looks good, but I did have a few units and namelist XML questions that I would like to get resolved before testing and merging.

Also @gold2718 if you feel like you won't have time to make any changes then I am happy to do them for you, as long as you are ok with me either pushing to your branch, or opening a PR.  Just let me know!

File: utilities/qneg.meta [hunk 1]
```diff
@@ -151,11 +175,29 @@
...
 [ isrootproc ]
-  standard_name = flag_for_mpi_root
+  standard_name = is_mpi_root
+  units = flag
+  type = logical
+  dimensions = ()
+  intent = in
+[ iulog ]
+  standard_name = log_output_unit
+  units = None
```
[the remark this answers is not in the document]
Comment (gold2718) [reply utilities/qneg.meta]: Agreed (and this was true even back in February).

File: utilities/qneg_namelist.xml [hunk 2]
```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0"?>
+
+<?xml-stylesheet type="text/xsl" href="namelist_definition.xsl"?>
```
[the remark this answers is not in the document]
Comment (gold2718) [reply utilities/qneg_namelist.xml]: It's the *old* 2-file format (currently only existing in CTSM)!
Removed the `href`, we should propagate this change around (e.g., to CDEPS which also has this useless appendage).

File: utilities/qneg_namelist.xml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+<entry_id_pg version="2.0">
+
+<!-- Each namelist variable is defined in an <entry> element.  The
+     content of the element is the documentation of how the variable is
+     used.  Other aspects of the variable's definition are expressed as
+     attributes of the <entry> element.  Note that it is an XML requirement
+     that the attribute values are enclosed in quotes.  The attributes are:
```
[the remark this answers is not in the document]
Comment (gold2718) [reply utilities/qneg_namelist.xml:11]: Great but use the new version.

File: utilities/qneg_namelist.xml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	  Any of these types may be followed by a comma separated list of
+	  integers enclosed in parenthesis to indicate an array.
+
+	  The current namelist validation code only distinquishes between
+	  string and non-string types.
+
+     input_pathname
+          Only include this attribute to indicate that the variable
+          contains the pathname of an input dataset that resides in the
+          CCSM inputdata directory tree.  Note that the variables
```
[the remark this answers is not in the document]
Comment (gold2718) [reply utilities/qneg_namelist.xml]: You have something against history?

File: utilities/qneg_namelist.xml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+     category
+          A category assigned for organizing the documentation.
+
+     group
+          The namelist group that the variable is declared in.
+
+     valid_values
+          This is an optional attribute that is mainly useful for variables
+          that have only a small number of allowed values.
+                                                                        -->
```
[the remark this answers is not in the document]
Comment (gold2718) [reply utilities/qneg_namelist.xml:77]: Good idea, added.

**Review by nusbaume — APPROVED** [review]
I am happy with this PR once gold2718#1 has been merged/resolved.

**Review by nusbaume — APPROVED** [review]

**Pull request merged**